internal/jobqueue/fsjobqueue/fsjobqueue.go

```diff
@@ -300,6 +304,28 @@ func (q *fsJobQueue) Job(id uuid.UUID) (jobType string, args json.RawMessage, de
 	return
 }
 
+func (q *fsJobQueue) IdFromToken(token uuid.UUID) (id uuid.UUID, err error) {
+	ids, err := q.db.List()
+	if err != nil {
+		return uuid.Nil, fmt.Errorf("error listing jobs: %v", err)
+	}
+
+	for _, id := range ids {
```

Comment:
This almost certainly does not matter, but linear search is making me a bit anxious ;) In principle the number of jobs could grow very large over time (we don't delete them when they finish), so reading them back in on every call might not be ideal (this may be mostly academic).

How about creating a second set of objects in addition to the jobs, that represent the "work"/"job slot" or whatever you want to call it. For now they could just be named after the token and contain the job id. That way lookup in both directions would be `O(1)`.

Reply:
Makes sense :)

Reply:
I think I can simplify it by keeping the token in the job struct, but just having a map of token -> id, populating that at the start when going through all jobs anyway, and then making sure that's threadsafe when reading or writing. No need to have another object on disk, as the fsqueue isn't meant to be used concurrently anyway, this should be fine I think?

Reply:
Yeah, that should be fine too!